From 5b890b47aaf6f692c876faf435e6bf8990e7a45d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ville=20Skytt=C3=A4?= <ville.skytta@iki.fi>
Date: Sat, 19 Jul 2014 21:06:58 +0300
Subject: [PATCH] compat.h needs stuff from config.h so include it there

This was brought up by dvdvml.* not including it and thus causing a
build failure on Fedora (22) devel. While at it, uninclude config.h
from files where it was included just for compat.h reasons.
---
 src/compat.h        | 2 ++
 src/conffile.c      | 1 -
 src/dvdauthor.c     | 2 --
 src/dvdcompile.c    | 2 --
 src/dvdpgc.c        | 1 -
 src/dvdunauthor.c   | 2 --
 src/dvduncompile.c  | 1 -
 src/dvdvob.c        | 2 --
 src/readxml.c       | 2 --
 src/spuunmux.c      | 1 -
 src/subgen-encode.c | 2 --
 11 files changed, 2 insertions(+), 16 deletions(-)

diff --git a/src/compat.h b/src/compat.h
index 126dcd7..93de081 100644
--- a/src/compat.h
+++ b/src/compat.h
@@ -1,6 +1,8 @@
 // basic headers
 #define _GNU_SOURCE /* really just for strndup */
 
+#include "config.h"
+
 #ifdef HAVE_STDBOOL_H
 # include <stdbool.h>
 #else
diff --git a/src/conffile.c b/src/conffile.c
index cab661a..36ded07 100644
--- a/src/conffile.c
+++ b/src/conffile.c
@@ -21,7 +21,6 @@
  * MA 02110-1301 USA.
  */
 
-#include "config.h"
 #include "compat.h"
 #include <errno.h>
 #include "conffile.h"
diff --git a/src/dvdauthor.c b/src/dvdauthor.c
index d591d68..4860258 100644
--- a/src/dvdauthor.c
+++ b/src/dvdauthor.c
@@ -20,8 +20,6 @@
  * MA 02110-1301 USA.
  */
 
-#include "config.h"
-
 #include "compat.h"
 
 #include <sys/types.h>
diff --git a/src/dvdcompile.c b/src/dvdcompile.c
index fd232b6..47f13e5 100644
--- a/src/dvdcompile.c
+++ b/src/dvdcompile.c
@@ -20,8 +20,6 @@
  * MA 02110-1301 USA.
  */
 
-#include "config.h"
-
 #include "compat.h"
 #include <assert.h>
 
diff --git a/src/dvdpgc.c b/src/dvdpgc.c
index e0adc61..cf6eb04 100644
--- a/src/dvdpgc.c
+++ b/src/dvdpgc.c
@@ -20,7 +20,6 @@
  * MA 02110-1301 USA.
  */
 
-#include "config.h"
 #include "compat.h"
 #include <errno.h>
 #include <assert.h>
diff --git a/src/dvdunauthor.c b/src/dvdunauthor.c
index ccd422f..f988cdf 100644
--- a/src/dvdunauthor.c
+++ b/src/dvdunauthor.c
@@ -23,8 +23,6 @@
  * MA 02110-1301 USA.
  */
 
-#include "config.h"
-
 #include "compat.h"
 
 #include <ctype.h>
diff --git a/src/dvduncompile.c b/src/dvduncompile.c
index 0c5eaf1..a064011 100644
--- a/src/dvduncompile.c
+++ b/src/dvduncompile.c
@@ -22,7 +22,6 @@
  * MA 02110-1301 USA.
  */
 
-#include "config.h"
 #include "compat.h"
 
 #include <stdio.h>
diff --git a/src/dvdvob.c b/src/dvdvob.c
index 2d7e2b8..2146689 100644
--- a/src/dvdvob.c
+++ b/src/dvdvob.c
@@ -20,8 +20,6 @@
  * MA 02110-1301 USA.
  */
 
-#include "config.h"
-
 #include "compat.h"
 
 #include <assert.h>
diff --git a/src/readxml.c b/src/readxml.c
index 61ff0a1..3b83f07 100644
--- a/src/readxml.c
+++ b/src/readxml.c
@@ -20,8 +20,6 @@
  * MA 02110-1301 USA.
  */
 
-#include "config.h"
-
 #include "compat.h"
 
 #include <assert.h>
diff --git a/src/spuunmux.c b/src/spuunmux.c
index 1981e41..94fcb10 100644
--- a/src/spuunmux.c
+++ b/src/spuunmux.c
@@ -31,7 +31,6 @@
  * MA 02110-1301 USA.
  */
 
-#include "config.h"
 #include "compat.h"
 
 #include <fcntl.h>
diff --git a/src/subgen-encode.c b/src/subgen-encode.c
index 53ed4b2..962e902 100644
--- a/src/subgen-encode.c
+++ b/src/subgen-encode.c
@@ -21,8 +21,6 @@
  * MA 02110-1301 USA.
  */
 
-#include "config.h"
-
 #include "compat.h"
 
 #include <assert.h>
 